### ISSUE TITLE
Request::cookies is empty if met  a 404 request

### ISSUE THREAD

**Issue by slince** [cakephp/cakephp#9611]
Hi guys

The  code like this:
```
//AppController::initialize()

 Log::info($this->request->cookies);
 Log::info($_COOKIE);
```
Request a 404 url:

![image](https://cloud.githubusercontent.com/assets/3785826/19349946/55c40624-9188-11e6-9e45-b1c3c65e7530.png)
System Log like:

![image](https://cloud.githubusercontent.com/assets/3785826/19349960/6fbd4d7e-9188-11e6-8b3b-28dfd0652628.png)

I will set a default "currency" cookie for client if couldn't find cookie; However if client request a 404 url;
"currency" cookie will be replaced with the default value.  it is unreasonable

**Comment by saeideng** [comment]
your cakephp version?

**Comment by slince** [comment]
@saeideng   3.3.4

**Comment by markstory** [comment]
I'll see if I can reproduce.

**Comment by slince** [comment]
@markstory  ok . In order to solve this， I will replace `$this->request->cookies` with "$_COOKIE" now

**Comment by saeideng** [comment]
please test latest version : 3.3.5
in this version some code changed around cookie

**Comment by BassemN** [comment]
@slince Try to test by running `composer require "cakephp/cakephp:dev-master as 3.3.6"`

**Comment by markstory** [comment]
I'm planning on packaging up 3.3.6 either tonight or on the weekend.

**Comment by slince** [comment]
@BassemN   @markstory   @saeideng    Nothing changed  in version 3.3.6

**Comment by markstory** [comment]
Ok. I'll see if I can reproduce then.

**Comment by markstory** [comment]
I wasn't able to reproduce this issue. I did the following:

1. Added some cookies with `response->cookie()`
2. Added the logging you have in `AppController::initialize()`
3. Ensured that `App\Controller\ErrorController` both extended `AppController` and called `parent::initialize()`.
4. Visited http://cake.localhost/not-a-valid-page in my browser. Both log messages contained the same cookie content.

I'll need a way to reproduce this issue before I can look into your issue further.
